tracer/src/Datadog.Trace.SourceGenerators/InstrumentationDefinitions/Sources.cs

```diff
@@ -22,8 +24,8 @@ namespace Datadog.Trace.ClrProfiler
 {
     internal static partial class InstrumentationDefinitions
     {
-        private static NativeCallTargetDefinition[] GetDefinitionsArray()
-            => new NativeCallTargetDefinition[]
+        private static InstrumentationDefinition[] GetDefinitionsArray()
```

Comment:
nit - could we make this code generic by enumerating the names in the enum? it's a nit cause the reading part in `InstrumentationDefinitions` will need to be coded anyway (:thinking_face: hmm maybe we could pass the value of the enum there as well and setting the payload on the enum as an attribute, but that may be over engineering).

Reply:
fixed in https://github.com/DataDog/dd-trace-dotnet/pull/2580/commits/2d5dc30a9ca98bded32867dbd88a320d175c2c88